packages/langium/src/validation/document-validator.ts

```diff
@@ -61,6 +61,9 @@ export class DefaultDocumentValidator implements DocumentValidator {
 
     async validateDocument(document: LangiumDocument, options: ValidationOptions = {}, cancelToken = CancellationToken.None): Promise<Diagnostic[]> {
         const parseResult = document.parseResult;
+        if (!parseResult.value.$cstNode) {
```

Comment:
Why the early-exit here?

Reply:
Because the validator requires the CST to identify where to place the validation markers. If I want to place a diagnostic on a keyword, the validator doesn't know where to place it, because all information about that keyword was lost when we discarded the CST.